rust/variable/variable_concept.rs

```diff
@@ -28,73 +28,56 @@ use crate::{
     variable::variable::validate_variable_name,
 };
 
-#[derive(Debug, Clone, Copy, Eq, Hash, PartialEq)]
-pub enum Visibility {
-    Visible,
-    Invisible,
-}
-
 #[derive(Debug, Clone, Eq, Hash, PartialEq)]
 pub enum ConceptVariable {
-    Anonymous(Visibility),
-    Name(String),
+    Anonymous,
+    Hidden,
+    Named(String),
 }
 
 impl ConceptVariable {
     const ANONYMOUS_NAME: &'static str = token::Char::Underscore.as_str();
 
-    pub fn named(name: String) -> ConceptVariable {
-        ConceptVariable::Name(name)
-    }
-
-    pub fn anonymous() -> ConceptVariable {
-        ConceptVariable::Anonymous(Visibility::Visible)
-    }
-
-    pub fn hidden() -> ConceptVariable {
-        ConceptVariable::Anonymous(Visibility::Invisible)
-    }
```

Comment:
These are not necessary, the variants can be constructed directly with near-identical syntax.